experimental/DoubleComplex/docs/src/user_interface.md

```diff
@@ -0,0 +1,83 @@
+```@meta
+CurrentModule = Oscar
+```
+# Double complexes -- the user's interface
+We briefly review the mathematical notion of a double complex. 
+Let ``\mathcal A`` be an Abelian category. A double complex 
+``D_{\bullet, \bullet}`` consists of a collection of objects ``D_{i, j}`` in 
+``\mathcal A`` with indices ``(i, j) \in \mathbb Z^2`` and usually arranged 
+in a matrix-like grid, together with two collections 
+of morphisms ``D_{i, j} \to D_{i \pm 1, j}``, the *horizontal* morphisms, and 
+``D_{i, j} \to D_{i, j \pm 1}``, the *vertical* morphisms, so that both 
+the rows and the columns of ``D_{\bullet, \bullet}`` are complexes in the 
+classical sense and such that all resulting squares of maps commute.
+
+In practice one usually encounters complexes which are *bounded* in the sense 
+that outside some specified area of indices ``(i, j) \in \mathbb Z^2`` the entries 
+``D_{i, j}`` are all zero. Such entries are then usually omitted. 
+
+## Basic getters and attributes
+In OSCAR the generic functionality for double complexes is declared for the 
+abstract type `AbsDoubleComplexOfMorphisms`. These functions comprise
+```julia
+  getindex(D::AbsDoubleComplexOfMorphisms, i::Int, j::Int) # Get the `(i,j)`-th entry of `D`
+```
+```@docs
+    horizontal_map(dc::AbsDoubleComplexOfMorphisms, i::Int, j::Int)
+    vertical_map(dc::AbsDoubleComplexOfMorphisms, i::Int, j::Int)
+```
+In which direction the maps in the rows and columns go can be asked with the following methods:
+```@docs
+    horizontal_direction(dc::AbsDoubleComplexOfMorphisms)
+    vertical_direction(dc::AbsDoubleComplexOfMorphisms)
+```
+Double complexes can be bounded or unbounded. But even in the bounded case where only finitely 
+many ``D_{i, j}`` can be non-zero, it is not clear that a concrete bound for the 
+range of indices ``(i, j)`` of possibly non-zero entries is known. Then again, even in that case 
+such a bound might be rather theoretical and it is still encouraged that double complexes be implemented 
+lazy and not fill out the full grid of morphisms on construction. Thus a merely theoretical bound 
+on the range of indices does not seem to be practically relevant. 
```

Comment:
The bounds do have practical relevance. Say you have a double complex  and you want to compute it's total complex. For every strand (direct sum of the `D_{i, j}` with `i + j = k`) there would be infinitely many pairs `(i, j)` that one would have to check to be zero. So bounding the range in which to look, is key here. 

The point here was to say that a *theoretical* bound is not relevant here. Only concretely known bounds are supported here, i.e. `has_upper_bound` is `true` only when a *concrete* bound is known (and not something like "We know a Groebner basis is finite, but it might also be a million elements"). 

I wanted to have all this discussion here so that people do not come forward saying: "Have you thought of this? And that?".

Reply:
I overhauled this part now.